include/deal.II/grid/grid_tools_cache_update_flags.h

```diff
@@ -0,0 +1,176 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2017 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE at
+// the top level of the deal.II distribution.
+//
+// ---------------------------------------------------------------------
+
+#ifndef dealii_grid_tria_info_cache_update_flags_h
+#define dealii_grid_tria_info_cache_update_flags_h
+
+
+#include <deal.II/base/config.h>
+
+DEAL_II_NAMESPACE_OPEN
+
+namespace GridTools
+{
+
+  /**
+   * The enum type given to the Cache class to select what
+   * information to update.
+   *
+   * You can select more than one flag by concatenation using the bitwise or
+   * <code>operator|(CacheUpdateFlags,CacheUpdateFlags)</code>.
+   *
+   * @author Luca Heltai, 2017.
+   */
+  enum CacheUpdateFlags
+  {
+    /**
+     * Update Nothing.
+     */
+    update_nothing = 0x00,
+
+    /**
+     * Update vertex_to_cell_map, as returned by GridTools::vertex_to_cell_map().
+     */
+    update_vertex_to_cell_map = 0x01,
+
+    /**
+     * Update vertex_to_cell_centers_directions, as returned by
+     * GridTools::vertex_to_cell_centers_directions()
+     */
+    update_vertex_to_cell_centers_directions = update_vertex_to_cell_map | 0x0002,
+
+#ifdef DEAL_II_WITH_NANOFLANN
+    /**
+     * Update a KDTree object, initialized with the vertices of the Triangulation.
+     */
+    update_vertex_kdtree = 0x04,
+#endif
```

Comment:
I think you want to make this flag available unconditionally and just throw a run-time error when the `get_vertex_kdtree()` function is called (which should also unconditionally be called). At least that's how we treat other dependencies such as UMFPACK.

Reply:
Will do the second part in a followup PR. This requires KDTree to be defined even when nanoflann is not available (which is not the case now. If nanoflann was not found, no KDTree class is defined.)

Reply:
Oh, I see.